pkg/agent/openflow/pipeline.go

```diff
@@ -1477,39 +1498,72 @@ func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svc
 	learnFlowBuilderLearnAction := learnFlowBuilder.
 		Action().Learn(sessionAffinityTable, priorityNormal, affinityTimeout, 0, cookieID).
 		DeleteLearned()
-	if protocol == binding.ProtocolTCP {
+	switch protocol {
+	case binding.ProtocolTCP:
 		learnFlowBuilder.MatchTCPDstPort(svcPort)
 		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedTCPDstPort()
-	} else if protocol == binding.ProtocolUDP {
+	case binding.ProtocolUDP:
 		learnFlowBuilder.MatchUDPDstPort(svcPort)
 		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedUDPDstPort()
-	} else if protocol == binding.ProtocolSCTP {
+	case binding.ProtocolSCTP:
 		learnFlowBuilder.MatchSCTPDstPort(svcPort)
 		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedSCTPDstPort()
+	case binding.ProtocolTCPv6:
+		learnFlowBuilder.MatchTCPv6DstPort(svcPort)
+		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedTCPv6DstPort()
+	case binding.ProtocolUDPv6:
+		learnFlowBuilder.MatchUDPv6DstPort(svcPort)
+		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedUDPv6DstPort()
+	case binding.ProtocolSCTPv6:
+		learnFlowBuilder.MatchSCTPv6DstPort(svcPort)
+		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedSCTPv6DstPort()
 	}
-	return learnFlowBuilderLearnAction.
-		MatchLearnedDstIP().
-		MatchLearnedSrcIP().
-		LoadRegToReg(int(endpointIPReg), int(endpointIPReg), endpointIPRegRange, endpointIPRegRange).
-		LoadRegToReg(int(endpointPortReg), int(endpointPortReg), endpointPortRegRange, endpointPortRegRange).
-		LoadReg(int(serviceLearnReg), marksRegServiceSelected, serviceLearnRegRange).
-		LoadReg(int(marksReg), macRewriteMark, macRewriteMarkRange).
-		Done().
-		Action().LoadRegRange(int(serviceLearnReg), marksRegServiceSelected, serviceLearnRegRange).
-		Action().GotoTable(endpointDNATTable).
-		Done()
+	ipProtocol := getIPProtocol(svcIP)
```

Comment:
I think the variable `ipProtocol` could be set value in the above switch-case according to the given protocol?

Reply:
Done